wlauto/common/android/device.py

```diff
@@ -649,6 +649,10 @@ def get_device_model(self):
         except KeyError:
             return None
 
+    def is_wifi_connected(self):
+        self.logger.debug('Checking for wi-fi connectivity.')
+        return self.execute('dumpsys wifi| grep enabled')
```

Comment:
I agree this is a more precise alternative. It reports correctly when the wifi adapter is on but not in a connected state.